### DISCUSSION TITLE
Support for HTML preview image on xchain.io

### DISCUSSION THREAD

**Discussion by babalicious-io** [CounterpartyXCP/CIPs#105] [CIPs]
I'd like to propose the idea to support preview images of .html files on the Counterparty Block Explorer.

I'm presuming this is a Stamps protocol specific request, as I haven't noticed html/css art on Counterparty (xcp).

It should be noted, that there may be security issues involved (since the html container can hold all kinds of code).

An Bitcoin Stamp example of html art (poetry).
https://xchain.io/asset/A6113378594505920000

As you can see, very little information about the asset/stamp is available.
It would definitely help this art niche evolve if users could see a preview image of the asset.

**Comment by jdogresorg** [comment]
1. The ability to associate HTML with a token via JSON already exists thanks to CIP25

https://xchain.io/asset/CIPXXV

2. XChain is a closed-source explorer developed to showcase COUNTERPARTY features... And the explorer already has the ability to associate HTML / CSS / Javascript, or any other type of content with the asset, and to have that content automatically loaded and displayed on xchain.io.

3.  Perhaps you should speak with the Bitcoin Stamps people (@mikeinspace) about HTML integration into stamps and building out their own block explorer with their own custom features to view their stamps.

**Reply by babalicious-io** [reply]
**Thank you for replying back and the link to the pepe puzzle game - fun idea !**
I'll have to dive more into html on counterparty at a later time.

I understand that xchain is counterparty specific, I just don't see why that necessarily has to exclude stamps ?!?
It will definitely benefit both subcultures (counterparty/stamps) if there is a technical infrastructure symbiosis/overlap, since there already seems to be on other aspects (human/artist/artistic).

Stamps have led me to a greater exploration of the counterparty ecosystem and art tokens, partly because of xchain explorer. 
If/when stamps build out their infrastructure and no longer rely on xchain/counterparty/freewallet, then you just loose potential onboarding to counterparty ! 
And vice versa.

Just sharing a perspective.

Best wishes !
